src/main/java/org/springframework/data/couchbase/config/AbstractCouchbaseConfiguration.java

```diff
@@ -130,6 +140,10 @@ public Cluster couchbaseCluster(ClusterEnvironment couchbaseClusterEnvironment)
 	@Bean(destroyMethod = "shutdown")
 	public ClusterEnvironment couchbaseClusterEnvironment() {
 		ClusterEnvironment.Builder builder = ClusterEnvironment.builder();
+		if (!nonShadowedJacksonPresent()) {
+			throw new CouchbaseException("non-shadowed Jackson not present");
+		}
+		builder.jsonSerializer(JacksonJsonSerializer.create(couchbaseObjectMapper));
```

Comment:
use the couchbaseObjectMapper from the new Bean.
Check for nonShadowedJacksonPresent() although it should never happen.